aunt-leahs-app/src/pages/newVolunteer.jsx

```diff
@@ -0,0 +1,118 @@
+      lname: "",
+      email: "",
+      address: "",
+      city: "",
+      province: "",
+      postal: "",
+      contactFName: "",
+      contactLName: "",
+      contactRelationship: "",
+      contactPhone: "",
+      contactEmail: "",
+      mailingList: false,
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleTickbox = this.handleTickbox.bind(this);
+    this.onClick = this.onClick.bind(this);
+  }
+
+  handleChange(event, stateToChange) {
+    this.setState({ [stateToChange] : event.target.value });
+  }
+
+  handleTickbox(event) {
+    this.setState({ mailingList : !this.state.mailingList });
+  }
+
+  onClick(event) {
+    console.log(this.state);
+  }
+
+  render() {
+    return (
+      <div className="new-volunteer">
+        <Head page={constants.HEADER.SUB_HEADER.signUp} />
+
+        <div className="new-volunteer-form">
+          <form>
+
+            <h3 className="new-volunteer-form-subheading">Basic Information</h3>
+            <div className="formRow">
+              <TextInput title="First name" size="Short" onChange={(e) => this.handleChange(e, 'fname')} value={this.state.fname} />
+              <TextInput title="Last name" size="Short" onChange={(e) => this.handleChange(e, 'lname')} value={this.state.lname} />
+            </div>
+            <TextInput title="Email" size="Long" />
+
+            <h3 className="new-volunteer-form-subheading">Address</h3>
+            <TextInput title="Street address" size="Long" onChange={(e) => this.handleChange(e, 'address')} value={this.state.address}/>
+            <div className="formRow">
+              <TextInput title="City" size="Short" onChange={(e) => this.handleChange(e, 'city')} value={this.state.city} />
+              <TextInput title="Province" size="Short" onChange={(e) => this.handleChange(e, 'province')} value={this.state.province} />
+            </div>
+            <TextInput title="Postal Code" size="Short" onChange={(e) => this.handleChange(e, 'postal')} value={this.state.postal}/>
+
+            <h3 className="new-volunteer-form-subheading">Emergency Contact</h3>
+            <div className="formRow">
+              <TextInput title="First name" size="Short" onChange={(e) => this.handleChange(e, 'contactFName')} value={this.state.contactFName}/>
+              <TextInput title="Last name" size="Short" onChange={(e) => this.handleChange(e, 'contactLName')} value={this.state.contactLName}/>
+            </div>
+            <div className="formRow">
+              <TextInput title="Relationship" size="Short" onChange={(e) => this.handleChange(e, 'contactRelationship')} value={this.state.contactRelationship}/>
+              <TextInput title="Phone number" size="Short" onChange={(e) => this.handleChange(e, 'contactPhone')} value={this.state.contactPhone} />
+            </div>
+            <TextInput title="Email Address" size="Long" onChange={(e) => this.handleChange(e, 'contactEmail')} value={this.state.contactPhone}/>
```

Comment:
Michael, this is fine for now, but you could write code that is far less repetitive if you used Material UI's Grid component and iterated over a form array that contains an object for each field. This form array could be imported from the constants file. Have a look at the docs for the Grid component here: https://material-ui.com/components/grid/ and here: https://material-ui.com/api/grid/. Consider making a ticket to refactor this code.